### ISSUE TITLE
QSyncthingTray does not connects to Syncthing v0.13.2

### ISSUE THREAD

**Issue by anuragm** [sieren/QSyncthingTray#115]
I am using the latest (0.4.7) release on OS X. When `Authentication` box is unchecked and `syncthing` gui has no authentication, QSyncthingTray fails to connect to the web server. Even when the we need no authentication, it tries to connect to `http://@:127.0.0.1:8384` and fails to connect.

If we do use authentication, and check the `Authentication` box and provide correct user and password, QSyncthingTray tries to connect to `http://user:pass@127.0.0.1:8384` and succeeds. 

In short, QSyncthingTray is not forming the correct connection URL when no authentication is required.

**Comment by anuragm** [comment]
I think the issue might be connected to Syncthing v0.13.2 . I upgraded my linux machine to v0.13.2, and now I can no longer connect to it as well (with password or otherwise).

**Comment by leprechau** [comment]
I had the same problem ... toggling "HTTPS for GUI" in the synching web interface and leaving the "http" url in QSyncthingTray seemed to resolve the issue.  It might be that the 307 redirect issued by synching is followed by QSyncthingTray to the correct location when connecting to the "http" address.

After enabling HTTPS for the GUI via the synching webui ...

```
charlie:~ ahurt$ curl -vvv http://localhost:8384/
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8384 (#0)
> GET / HTTP/1.1
> Host: localhost:8384
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 307 Temporary Redirect
< Location: https://localhost:8384/
< Date: Sun, 22 May 2016 03:37:44 GMT
< Content-Length: 59
< Content-Type: text/html; charset=utf-8
<
<a href="https://localhost:8384/">Temporary Redirect</a>.

* Connection #0 to host localhost left intact
```

**Comment by anuragm** [comment]
@leprechau, your suggested workaround works 👍 .

**Comment by leprechau** [comment]
I was also able to get it to stop including the "@:"  with a simple patch to the ```SyncConnector::setURL``` function.  This made it form the correct URL when not passing authentication but it still fails to connect without the HTTPS workaround.  Maybe @sieren can figure it out.

```diff
diff --git a/src/syncconnector.cpp b/src/syncconnector.cpp
index 38e1e62..75e5707 100644
--- a/src/syncconnector.cpp
+++ b/src/syncconnector.cpp
@@ -63,8 +63,10 @@ void SyncConnector::setURL(QUrl url, std::string username, std::string password,
   ConnectionStateCallback setText)
 {
   mAuthentication = std::make_pair(username, password);
-  url.setUserName(mAuthentication.first.c_str());
-  url.setPassword(mAuthentication.second.c_str());
+  if (username.empty() == false && password.empty() == false) {
+    url.setUserName(mAuthentication.first.c_str());
+    url.setPassword(mAuthentication.second.c_str());
+  }
   mCurrentUrl = url;
   mConnectionStateCallback = setText;
   testUrlAvailability();
```

**Comment by sieren** [comment]
Thanks for the reports everyone. I was gone this weekend so I couldn't look into this, but almost certainly something in the v13 version of Syncthing must be handled differently now. I'll have a look this week.

**Comment by meztup** [comment]
On Win 7 x64, the workaround doesn't really work (although it does connect). Possibly due to API changes in Syncthing 0.13.2, QST can't get any info from it to animate the icon, display stats, or send the shutdown message on exit.

**Comment by sieren** [comment]
Yeah, there are a few changes in the recent version that broke a lot of things. See referenced issues.

**Comment by Getron** [comment]
Hi

The same thing here on Netrunner Rolling 2016.01 Arch KDE release Linux, it connects, i'm able to see the GUI, but, on preferences, it shows error:

Status: Error transfering https://..... (my authentication on https)

I have tryed with and without https, and with and without authentication (changed on Syncthing GUI first), but no luck, the error still there, and the infos about up and download are not showing and the animation is not showing, just a gray icon on tray (monochrome icon on settings is not picked). These apears just if not connected, or syncthing is not started on my machine.

I have Syncthing v0.13.2

**Comment by sieren** [comment]
Hey everyone,
quick update, there's now a branch/PR with various fixes and a working V13 implementation:
https://github.com/sieren/QSyncthingTray/pull/121

I'll hopefully merge it tomorrow, as I still need to validate the correct Syncthing Configuration Directories on Windows and Linux.
So far I can only confirm it's working on Mac.

**Comment by sieren** [comment]
Hey everyone,

there's a new QST release out now with fixes to the aforementioned issues.
Let me know how this works for you:
https://github.com/sieren/QSyncthingTray/releases/tag/0.5.0

**Comment by Getron** [comment]
@sieren Hi Matt, it works great now. I have tested it, perfect.

Maybe, we should use a bigger window for showing the gui. Like now, it is looks like would open the gui in a smart phone, because the size of window is too small. Maybe we can set the window size bigger, to show entire GUI, like in browser?

**Comment by meztup** [comment]
Working great in Windows land also!

I could get behind an option to remember our gui window size...

**Comment by anuragm** [comment]
@sieren The new update works for me. Feel free to close the issue.

**Comment by sieren** [comment]
Good to know.. :)
Added an issue for the window request.
